### PR TITLE
feature(CCLS-1974) proceedings and costs amendments for ebs-api

### PR DESCRIPTION
**Pull request by PhilDigitalJustice**
Includes:
- CCLS-2044 amended proceedings endpoint for lead proceedings
- CCLS-2045 client involvement type endpoint
- CCLS-2046 level of service endpoint
- CCLS-2043 matter types endpoint

### CODE REVIEW

File: data-service/src/main/java/uk/gov/laa/ccms/data/controller/ProceedingController.java [hunk 1]
```diff
@@ -30,24 +30,44 @@ public class ProceedingController implements ProceedingsApi {
...
+      final String categoryOfLaw,
+      final String matterType,
+      final Boolean amendmentOnly,
+      final Boolean enabled,
+      final Boolean lead,
+      final String applicationType,
+      final Boolean larScopeFlag,
+      final Pageable pageable) {
+
+    if (lead != null && lead) {
```
Comment (porritta) [inline data-service/src/main/java/uk/gov/laa/ccms/data/controller/ProceedingController.java:51]: Boolean.TRUE.equals(lead), saves checking for null

File: data-service/src/main/java/uk/gov/laa/ccms/data/entity/LevelOfServiceId.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  @Column(name = "MATTER_TYPE")
+  private String matterType;
+
+  /**
+   * The proceeding code.
+   */
+  @Column(name = "PROCEEDING_CODE")
+  private String proceedingCode;
+
+  @Column(name = "LEVEL_OF_SERVICE_CODE")
```
Comment (porritta) [inline data-service/src/main/java/uk/gov/laa/ccms/data/entity/LevelOfServiceId.java:36]: javadoc

File: data-service/src/main/java/uk/gov/laa/ccms/data/entity/MatterType.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ */
+@Entity
+@Data
+@NoArgsConstructor
+@Table(name = "XXCCMS_MATTER_TYPES_V")
+@Immutable
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+public class MatterType {
+
+  @Id
```
Comment (porritta) [inline data-service/src/main/java/uk/gov/laa/ccms/data/entity/MatterType.java:25]: javadocs for attributes

File: data-service/src/main/java/uk/gov/laa/ccms/data/entity/ProceedingClientInvolvementTypeId.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+public class ProceedingClientInvolvementTypeId implements Serializable {
+
+  /**
+   * The proceeding code.
+   */
+  @Column(name = "PROCEEDING_CODE")
+  private String proceedingCode;
+
+  /**
+   * The stage end value.
```
Comment (porritta) [inline data-service/src/main/java/uk/gov/laa/ccms/data/entity/ProceedingClientInvolvementTypeId.java:25]: copy paste error

File: data-service/src/main/java/uk/gov/laa/ccms/data/service/ProceedingService.java [hunk 5]
```diff
@@ -54,6 +56,43 @@ public ProceedingDetails getProceedings(String categoryOfLawCode, String matterT
...
+  public ProceedingDetails getLeadProceedings(
+      String categoryOfLaw,
+      String matterType,
+      Boolean amendmentOnly,
+      Boolean enabled,
+      String applicationType,
+      Boolean larScopeFlag,
+      Pageable pageable) {
+
+    String amendmentOnlyString = (amendmentOnly != null) ? (amendmentOnly ? "Y" : "N") : null;
```
Comment (porritta) [inline data-service/src/main/java/uk/gov/laa/ccms/data/service/ProceedingService.java:81]: Could use our existing BooleanConverter.convertToDatabaseColumn

**Review by porritta — CHANGES_REQUESTED** [review]
comments added

File: data-service/src/main/java/uk/gov/laa/ccms/data/controller/ProceedingController.java (hunk 1 above)
Comment (PhilDigitalJustice) [reply data-service/src/main/java/uk/gov/laa/ccms/data/controller/ProceedingController.java:51]: amended

File: data-service/src/main/java/uk/gov/laa/ccms/data/entity/LevelOfServiceId.java (hunk 2 above)
Comment (PhilDigitalJustice) [reply data-service/src/main/java/uk/gov/laa/ccms/data/entity/LevelOfServiceId.java:36]: amended

File: data-service/src/main/java/uk/gov/laa/ccms/data/entity/ProceedingClientInvolvementTypeId.java (hunk 4 above)
Comment (PhilDigitalJustice) [reply data-service/src/main/java/uk/gov/laa/ccms/data/entity/ProceedingClientInvolvementTypeId.java:25]: amended

File: data-service/src/main/java/uk/gov/laa/ccms/data/service/ProceedingService.java (hunk 5 above)
Comment (PhilDigitalJustice) [reply data-service/src/main/java/uk/gov/laa/ccms/data/service/ProceedingService.java:81]: done

**Review by porritta — APPROVED** [review]
LGTM

**Pull request merged**